playbooks/setup_openstack_instances.yml

```diff
@@ -58,16 +76,74 @@
       loop_control:
         loop_var: instance
 
-    - name: Lock instances to prevent changes
+    - name: Create data volumes
+      os_volume:
+        display_name: "{{ item[1]['name'] }}"
+        volume_type: "{{ item[1]['type'] }}"
+        size: "{{ item[1]['size'] }}"
+        state: present
+        cloud: "{{ cloud_name | default('public_cloud') }}"
+        region_name: "{{ item[0]['region'] }}"
+        wait: yes
+        timeout: 900
+      when:
+        - "{{ item[0]['state'] | default('present') != 'absent' }}"
+      with_subelements:
+        - "{{ instance_list | default([]) }}"
+        - volumes
+        - skip_missing: True
+
+    - name: Delete data volumes on deleted instances
+      os_volume:
+        display_name: "{{ item[1]['name'] }}"
+        volume_type: "{{ item[1]['type'] }}"
+        size: "{{ item[1]['size'] }}"
+        state: absent
+        cloud: "{{ cloud_name | default('public_cloud') }}"
+        region_name: "{{ item[0]['region'] }}"
+        wait: yes
+        timeout: 900
+      when:
+        - "{{ item[0]['state'] | default('present') == 'absent' and item[1]['state'] | default('present') == 'absent' }}"
```

Comment:
if you put that in two lines it improves readablity.

Reply:
Agreed, updating.